src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_seckey.c

```diff
@@ -139,3 +139,68 @@ uint64_t AppleCryptoNative_SecKeyGetSimpleKeySizeInBytes(SecKeyRef publicKey)
 
     return SecKeyGetBlockSize(publicKey);
 }
+
+static CFStringRef GetKeyAlgorithmIdentifier(PAL_KeyAlgorithm keyAlgorithm)
+{
+    if (keyAlgorithm == PAL_KeyAlgorithm_EC)
+        return kSecAttrKeyTypeECSECPrimeRandom;
+    if (keyAlgorithm == PAL_KeyAlgorithm_RSA)
+        return kSecAttrKeyTypeRSA;
+    return NULL;
+}
+
+int32_t AppleCryptoNative_SecKeyCreateWithData(uint8_t* pKey,
+                                               int32_t cbKey,
+                                               PAL_KeyAlgorithm keyAlgorithm,
+                                               int32_t isPublic,
+                                               SecKeyRef* pKeyOut,
+                                               CFErrorRef* pErrorOut)
+{
+    assert(pKey != NULL);
+    assert(cbKey > 0);
+    assert(pKeyOut != NULL);
+    assert(pErrorOut != NULL);
```

Comment:
There are return paths that don't assign *pKeyOut or *pErrorOut.  Please write NULL to them before doing any other functional code, so we don't end up in a situation where the runtime didn't clear a local and now we're tracking it as a SafeHandle.